app/javascript/components/AdminProgram.tsx

```diff
@@ -0,0 +1,167 @@
+              anarchy
+            </button>
+          </Section>
+          <Section name="Difficulty">
+            <label>Max Char Length</label>
+            <div>
+              <button
+                onClick={() => channel.setMaxInputMode(MaxInputMode.Char)}
+                disabled={program.settings.max_input_mode == MaxInputMode.Char}
+              >
+                Character
+              </button>
+              <button
+                onClick={() => channel.setMaxInputMode(MaxInputMode.Word)}
+                disabled={program.settings.max_input_mode == MaxInputMode.Word}
+              >
+                Word
+              </button>
+              <button
+                onClick={() => channel.setMaxInputMode(MaxInputMode.Line)}
+                disabled={program.settings.max_input_mode == MaxInputMode.Line}
+              >
+                Line
+              </button>
+            </div>
+            <br />
+            <label>Vote Interval</label>
+            <div>
+              <button onClick={() => channel.setVoteInterval(1)} disabled={program.settings.vote_interval == 1}>
+                1 sec
+              </button>
+              <button onClick={() => channel.setVoteInterval(3)} disabled={program.settings.vote_interval == 3}>
+                3 sec
+              </button>
+              <button onClick={() => channel.setVoteInterval(5)} disabled={program.settings.vote_interval == 5}>
+                5 sec
+              </button>
+              <button onClick={() => channel.setVoteInterval(9)} disabled={program.settings.vote_interval == 9}>
+                9 sec
+              </button>
+            </div>
+          </Section>
+          <Section name={`Play Controls: (${program.settings.play_state || "playing"})`}>
+            <PlayControls channel={channel} current={program.settings.play_state as PlayState} />
+          </Section>
+          <Section name="Restart Controls">
+            <button onClick={channel.clear}>Clear Progress</button>
+            <button onClick={channel.resetTick}>Reset Timer</button>
+            <button onClick={channel.reset}>Restart</button>
+          </Section>
+        </div>
+      </div>
+    </>
+  );
+};
+
+const Section = ({ name, children }) => (
+  <section>
+    <h3>{name}</h3>
+    {children}
+    <hr />
+  </section>
+);
+
+interface PlayControlProps {
+  current: PlayState;
+  channel: ProgramChannel;
+}
+type PlayState = "playing" | "paused";
+const PlayControls = ({ channel, current }: PlayControlProps) => {
+  switch (current) {
+    case "paused":
+      return <button onClick={channel.resume}>Resume</button>;
+    case "playing":
+    default:
+      return <button onClick={channel.pause}>Pause</button>;
+  }
+};
+
+const tickAsTime = (tick: number): string => [leftPad(Math.floor(tick / 60)), ":", leftPad(tick % 60)].join("");
+
+const leftPad = (number: number): string => String(number).padStart(2, "0");
+
+export default AdminProgram;
```

Comment:
This entire AdminProgram is just a rough copy/paste of the original Program, now that both the admin and user views are getting close to being "feature complete", we can look for areas to pull out common components between the two interfaces.

I also need to go back through both Programs and inject a wrapper class to have better data fetch and presentation separation.